### ISSUE TITLE
Rule suggestion: no-invalid-string-templates

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#2283]
I common mistake I sometimes see is when someone forgets to use the backtick characters for a string template, but they still use the `${foo}` syntax inside the non-template string.

**Invalid Example** (uses regular quotes, but contains a template string formatter)
```ts
if (selectedElement == undefined) {
    throw Error("An element was not found using the selector '${selector}'");
   //                                                         ^^^^^^^^^^^ - throw error here
}
```

**Correct Example** (Correctly uses the backtick characters around the template string formatters)
```ts
if (selectedElement == undefined) {
    throw Error(`An element was not found using the selector '${selector}'`);
}
```
